Errors while installing the plugin + Aquara door/window contact still doesn't work.

There are still some errors occurring while installing the plugin in Homebridge.

Also I still can't add my Xiaomi Aquara Door/Window sensor. The issue is all the same as my lastly reported issue.

I've reinstalled the plugin multiple times and tried a bunch of different devices.

I hope you can help.

prebuild-install WARN install No prebuilt binaries found (target=12.18.3 runtime=node arch=arm libc= platform=linux)
make: Verzeichnis „/home/pi/node_modules/@serialport/bindings/build“ wird betreten
  CXX(target) Release/obj.target/bindings/src/serialport.o
In file included from ../src/./serialport.h:6,
                 from ../src/serialport.cpp:1:
../../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../../nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Open(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:78:69: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Open, (uv_after_work_cb)EIO_AfterOpen);
                                                                     ^~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Update(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:135:71: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Update, (uv_after_work_cb)EIO_AfterUpdate);
                                                                       ^~~~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Close(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:175:70: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Close, (uv_after_work_cb)EIO_AfterClose);
                                                                      ^~~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Flush(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:215:70: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Flush, (uv_after_work_cb)EIO_AfterFlush);
                                                                      ^~~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Set(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:270:68: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Set, (uv_after_work_cb)EIO_AfterSet);
                                                                    ^~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Get(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:314:68: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Get, (uv_after_work_cb)EIO_AfterGet);
                                                                    ^~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE GetBaudRate(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:363:76: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_GetBaudRate, (uv_after_work_cb)EIO_AfterGetBaudRate);
                                                                            ^~~~~~~~~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Drain(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:409:70: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Drain, (uv_after_work_cb)EIO_AfterDrain);
                                                                      ^~~~~~~~~~~~~~
../src/serialport.cpp: At global scope:
../src/serialport.cpp:430:28: warning: unnecessary parentheses in declaration of ‘ToParityEnum’ [-Wparentheses]
 SerialPortParity NAN_INLINE(ToParityEnum(const v8::Local<v8::String>& v8str)) {
                            ^
../src/serialport.cpp:449:30: warning: unnecessary parentheses in declaration of ‘ToStopBitEnum’ [-Wparentheses]
 SerialPortStopBits NAN_INLINE(ToStopBitEnum(double stopBits)) {
                              ^
In file included from ../../../nan/nan.h:56,
                 from ../src/./serialport.h:6,
                 from ../src/serialport.cpp:1:
/home/pi/.cache/node-gyp/12.18.3/include/node/node.h:608:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
       (node::addon_register_func) (regfunc),                          \
                                           ^
/home/pi/.cache/node-gyp/12.18.3/include/node/node.h:642:3: note: in expansion of macro ‘NODE_MODULE_X’
   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
   ^~~~~~~~~~~~~
../src/serialport.cpp:483:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(serialport, init);
 ^~~~~~~~~~~
  CXX(target) Release/obj.target/bindings/src/serialport_unix.o
In file included from ../src/serialport.h:6,
                 from ../src/serialport_unix.cpp:2:
../../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../../nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/serialport_unix.cpp: In function ‘int setup(int, OpenBaton*)’:
../src/serialport_unix.cpp:176:60: warning: ‘%s’ directive output may be truncated writing up to 1023 bytes into a region of size 1005 [-Wformat-truncation=]
     snprintf(data->errorString, sizeof(data->errorString), "Error %s Cannot open %s", strerror(errno), data->path);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~
../src/serialport_unix.cpp:176:13: note: ‘snprintf’ output 20 or more bytes (assuming 1043) into a destination of size 1024
     snprintf(data->errorString, sizeof(data->errorString), "Error %s Cannot open %s", strerror(errno), data->path);
     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/serialport_unix.cpp: In function ‘void EIO_Open(uv_work_t*)’:
../src/serialport_unix.cpp:86:60: warning: ‘%s’ directive output may be truncated writing up to 1023 bytes into a region of size 1003 [-Wformat-truncation=]
     snprintf(data->errorString, sizeof(data->errorString), "Error: %s, cannot open %s", strerror(errno), data->path);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/serialport_unix.cpp:86:13: note: ‘snprintf’ output 22 or more bytes (assuming 1045) into a destination of size 1024
     snprintf(data->errorString, sizeof(data->errorString), "Error: %s, cannot open %s", strerror(errno), data->path);
     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CXX(target) Release/obj.target/bindings/src/poller.o
In file included from ../src/poller.cpp:1:
../../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../../nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
  CXX(target) Release/obj.target/bindings/src/serialport_linux.o
  SOLINK_MODULE(target) Release/obj.target/bindings.node
  COPY Release/bindings.node
make: Verzeichnis „/home/pi/node_modules/@serialport/bindings/build“ wird verlassen

> @serialport/bindings@9.0.0 install /home/pi/node_modules/zigbee-herdsman/node_modules/@serialport/bindings
> prebuild-install --tag-prefix @serialport/bindings@ || node-gyp rebuild

prebuild-install WARN install No prebuilt binaries found (target=12.18.3 runtime=node arch=arm libc= platform=linux)
make: Verzeichnis „/home/pi/node_modules/zigbee-herdsman/node_modules/@serialport/bindings/build“ wird betreten
  CXX(target) Release/obj.target/bindings/src/serialport.o
In file included from ../src/./serialport.h:6,
                 from ../src/serialport.cpp:1:
../../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../../nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Open(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:78:69: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Open, (uv_after_work_cb)EIO_AfterOpen);
                                                                     ^~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Update(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:135:71: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Update, (uv_after_work_cb)EIO_AfterUpdate);
                                                                       ^~~~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Close(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:175:70: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Close, (uv_after_work_cb)EIO_AfterClose);
                                                                      ^~~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Flush(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:215:70: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Flush, (uv_after_work_cb)EIO_AfterFlush);
                                                                      ^~~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Set(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:270:68: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Set, (uv_after_work_cb)EIO_AfterSet);
                                                                    ^~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Get(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:314:68: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Get, (uv_after_work_cb)EIO_AfterGet);
                                                                    ^~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE GetBaudRate(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:363:76: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_GetBaudRate, (uv_after_work_cb)EIO_AfterGetBaudRate);
                                                                            ^~~~~~~~~~~~~~~~~~~~
../src/serialport.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE Drain(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/serialport.cpp:409:70: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
   uv_queue_work(uv_default_loop(), req, EIO_Drain, (uv_after_work_cb)EIO_AfterDrain);
                                                                      ^~~~~~~~~~~~~~
../src/serialport.cpp: At global scope:
../src/serialport.cpp:430:28: warning: unnecessary parentheses in declaration of ‘ToParityEnum’ [-Wparentheses]
 SerialPortParity NAN_INLINE(ToParityEnum(const v8::Local<v8::String>& v8str)) {
                            ^
../src/serialport.cpp:449:30: warning: unnecessary parentheses in declaration of ‘ToStopBitEnum’ [-Wparentheses]
 SerialPortStopBits NAN_INLINE(ToStopBitEnum(double stopBits)) {
                              ^
In file included from ../../../nan/nan.h:56,
                 from ../src/./serialport.h:6,
                 from ../src/serialport.cpp:1:
/home/pi/.cache/node-gyp/12.18.3/include/node/node.h:608:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
       (node::addon_register_func) (regfunc),                          \
                                           ^
/home/pi/.cache/node-gyp/12.18.3/include/node/node.h:642:3: note: in expansion of macro ‘NODE_MODULE_X’
   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
   ^~~~~~~~~~~~~
../src/serialport.cpp:483:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(serialport, init);
 ^~~~~~~~~~~
  CXX(target) Release/obj.target/bindings/src/serialport_unix.o
In file included from ../src/serialport.h:6,
                 from ../src/serialport_unix.cpp:2:
../../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../../nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/serialport_unix.cpp: In function ‘int setup(int, OpenBaton*)’:
../src/serialport_unix.cpp:176:60: warning: ‘%s’ directive output may be truncated writing up to 1023 bytes into a region of size 1005 [-Wformat-truncation=]
     snprintf(data->errorString, sizeof(data->errorString), "Error %s Cannot open %s", strerror(errno), data->path);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~
../src/serialport_unix.cpp:176:13: note: ‘snprintf’ output 20 or more bytes (assuming 1043) into a destination of size 1024
     snprintf(data->errorString, sizeof(data->errorString), "Error %s Cannot open %s", strerror(errno), data->path);
     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/serialport_unix.cpp: In function ‘void EIO_Open(uv_work_t*)’:
../src/serialport_unix.cpp:86:60: warning: ‘%s’ directive output may be truncated writing up to 1023 bytes into a region of size 1003 [-Wformat-truncation=]
     snprintf(data->errorString, sizeof(data->errorString), "Error: %s, cannot open %s", strerror(errno), data->path);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/serialport_unix.cpp:86:13: note: ‘snprintf’ output 22 or more bytes (assuming 1045) into a destination of size 1024
     snprintf(data->errorString, sizeof(data->errorString), "Error: %s, cannot open %s", strerror(errno), data->path);
     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CXX(target) Release/obj.target/bindings/src/poller.o
In file included from ../src/poller.cpp:1:
../../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../../nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
  CXX(target) Release/obj.target/bindings/src/serialport_linux.o
  SOLINK_MODULE(target) Release/obj.target/bindings.node
  COPY Release/bindings.node
make: Verzeichnis „/home/pi/node_modules/zigbee-herdsman/node_modules/@serialport/bindings/build“ wird verlassen

> serialport@9.0.0 postinstall /home/pi/node_modules/zigbee-herdsman/node_modules/serialport
> node thank-you.js

Thank you for using serialport!
If you rely on this package, please consider supporting our open collective:
> https://opencollective.com/serialport/donate



> serialport@8.0.8 postinstall /home/pi/node_modules/serialport
> node thank-you.js

--- a/config.schema.json
+++ b/config.schema.json
@@ -78,11 +78,12 @@
         "required": false
       },
       "routerPollingInterval": {
-        "title": "Router Polling Interval",
-        "description": "Router polling interval, some routers need polling to prevent them from sleeping (in seconds).",
+        "title": "Ping Interval",
+        "description": "Ping interval for routers to prevent them from sleeping (in seconds).",
         "type": "number",
-        "default": 120,
+        "default": 30,
         "minimum": 10,
+        "maximum": 180,
         "required": false
       },
       "disableLed": {
@@ -90,13 +91,6 @@
         "description": "Turn off the stick led indicator.",
         "type": "boolean",
         "default": false,
-        "required": false
-      },
-      "disablePingLog": {
-        "title": "Disable Ping Log",
-        "description": "Disable ping records in log.",
-        "type": "boolean",
-        "default": true,
         "required": false
       }
     }
@@ -109,7 +103,6 @@
     "secondaryChannel",
     "database",
     "routerPollingInterval",
-    "disablePingLog",
     "disableLed"
   ]
 }

--- a/config.schema.json
+++ b/config.schema.json
@@ -108,9 +108,8 @@
     "channel",
     "secondaryChannel",
     "database",
-    "permitJoinTimeout",
     "routerPollingInterval",
-    "disableLed",
-    "disablePingLog"
+    "disablePingLog",
+    "disableLed"
   ]
 }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "homebridge-zigbee-nt",
-  "version": "1.0.2",
+  "version": "1.0.3",
   "description": "ZigBee New Technology Platform plugin for HomeBridge",
   "main": "dist/index.js",
   "scripts": {

--- a/src/accessories/zig-bee-accessory.ts
+++ b/src/accessories/zig-bee-accessory.ts
@@ -106,10 +106,10 @@ export abstract class ZigBeeAccessory {
     }
   }
 
-  async update(device: Device, state: DeviceState) {
+  update(device: Device, state: DeviceState) {
     Object.assign(this.accessory.context, { ...device });
     Object.assign(this.state, state);
     this.zigBeeDeviceDescriptor.updateLastSeen();
-    await this.configureDevice();
+    this.configureDevice();
   }
 }

--- a/src/accessories/zig-bee-accessory.ts
+++ b/src/accessories/zig-bee-accessory.ts
@@ -4,6 +4,7 @@ import { ZigBeeClient } from '../zigbee/zig-bee-client';
 import { findByDevice } from 'zigbee-herdsman-converters';
 import { Device, Endpoint } from 'zigbee-herdsman/dist/controller/model';
 import { DeviceState, ZigBeeDefinition, ZigBeeEntity } from '../zigbee/types';
+import { isDeviceRouter } from '../utils/router-polling';
 
 export interface ZigBeeAccessoryCtor {
   new (
@@ -69,35 +70,37 @@ export abstract class ZigBeeAccessory {
   async ping() {
     try {
       await this.zigBeeDeviceDescriptor.ping();
-      if (this.isConfigured === false) {
-        await this.configureDevice();
-        this.isConfigured = true;
-      }
+      await this.configureDevice();
     } catch (e) {
-      this.log.error(`No response from ${this.zigBeeDefinition.description}. Reset configuration.`);
-      this.isConfigured = false;
+      this.log.warn(`No response from ${this.zigBeeDefinition.description}. Is it online?`);
     }
   }
 
   public abstract getAvailableServices(): Service[];
 
   public async onDeviceMount() {
-    await this.ping();
-    setInterval(() => {
-      this.ping();
-    }, 30000);
+    this.zigBeeDeviceDescriptor.updateLastSeen();
+    if (isDeviceRouter(this.zigBeeDeviceDescriptor)) {
+      await this.ping();
+      setInterval(() => {
+        this.ping();
+      }, 30000);
+    } else {
+      await this.configureDevice();
+    }
   }
 
   public async configureDevice() {
-    this.zigBeeDeviceDescriptor.updateLastSeen();
-    if (this.entity.definition.configure) {
+    if (this.entity.definition.configure && !this.isConfigured) {
       try {
         await this.entity.definition.configure(
           this.zigBeeDeviceDescriptor,
           this.coordinatorEndpoint
         );
+        this.isConfigured = true;
       } catch (e) {
         this.log.error(`Cannot configure device ${this.name}: ${e.message}`);
+        this.isConfigured = false;
       }
     }
   }

--- a/src/accessories/zig-bee-accessory.ts
+++ b/src/accessories/zig-bee-accessory.ts
@@ -4,7 +4,12 @@ import { ZigBeeClient } from '../zigbee/zig-bee-client';
 import { findByDevice } from 'zigbee-herdsman-converters';
 import { Device, Endpoint } from 'zigbee-herdsman/dist/controller/model';
 import { DeviceState, ZigBeeDefinition, ZigBeeEntity } from '../zigbee/types';
-import { isDeviceRouter } from '../utils/router-polling';
+import {
+  DEFAULT_POLL_INTERVAL,
+  isDeviceRouter,
+  MAX_POLL_INTERVAL,
+  MIN_POLL_INTERVAL,
+} from '../utils/router-polling';
 
 export interface ZigBeeAccessoryCtor {
   new (
@@ -79,12 +84,18 @@ export abstract class ZigBeeAccessory {
   public abstract getAvailableServices(): Service[];
 
   public async onDeviceMount() {
+    this.log.info(`Mounting device ${this.name}...`);
     this.zigBeeDeviceDescriptor.updateLastSeen();
     if (isDeviceRouter(this.zigBeeDeviceDescriptor)) {
+      this.log.info(`Device ${this.name} is a router, install ping`);
+      let interval = this.platform.config.routerPollingInterval * 1000 || DEFAULT_POLL_INTERVAL;
+      if (interval < MIN_POLL_INTERVAL || interval > MAX_POLL_INTERVAL) {
+        interval = DEFAULT_POLL_INTERVAL;
+      }
       await this.ping();
       setInterval(() => {
         this.ping();
-      }, 30000);
+      }, interval);
     } else {
       await this.configureDevice();
     }
@@ -107,9 +118,10 @@ export abstract class ZigBeeAccessory {
   }
 
   update(device: Device, state: DeviceState) {
+    this.log.debug(`Updating state of device ${this.name}: `, state);
     Object.assign(this.accessory.context, { ...device });
     Object.assign(this.state, state);
     this.zigBeeDeviceDescriptor.updateLastSeen();
-    this.configureDevice();
+    this.configureDevice(); // Ignore the promise result (try/catch inside the function)
   }
 }

--- a/src/accessories/zig-bee-accessory.ts
+++ b/src/accessories/zig-bee-accessory.ts
@@ -22,7 +22,8 @@ export abstract class ZigBeeAccessory {
   protected readonly client: ZigBeeClient;
   protected state: DeviceState;
   protected readonly entity: ZigBeeEntity;
-  private coordinatorEndpoint: Endpoint;
+  private readonly coordinatorEndpoint: Endpoint;
+  private isConfigured: boolean;
 
   constructor(
     platform: ZigbeeNTHomebridgePlatform,
@@ -39,6 +40,7 @@ export abstract class ZigBeeAccessory {
     this.accessory.context = device;
     this.entity = this.client.resolveEntity(device);
     this.coordinatorEndpoint = this.client.getCoodinator().getEndpoint(1);
+    this.isConfigured = false;
     let Characteristic = platform.Characteristic;
     this.accessory
       .getService(this.platform.Service.AccessoryInformation)
@@ -64,9 +66,29 @@ export abstract class ZigBeeAccessory {
     return this.zigBeeDefinition?.description;
   }
 
+  async ping() {
+    try {
+      await this.zigBeeDeviceDescriptor.ping();
+      if (this.isConfigured === false) {
+        await this.configureDevice();
+        this.isConfigured = true;
+      }
+    } catch (e) {
+      this.log.error(`No response from ${this.zigBeeDefinition.description}. Reset configuration.`);
+      this.isConfigured = false;
+    }
+  }
+
   public abstract getAvailableServices(): Service[];
 
   public async onDeviceMount() {
+    await this.ping();
+    setInterval(() => {
+      this.ping();
+    }, 30000);
+  }
+
+  public async configureDevice() {
     this.zigBeeDeviceDescriptor.updateLastSeen();
     if (this.entity.definition.configure) {
       try {

--- a/src/accessories/zig-bee-accessory.ts
+++ b/src/accessories/zig-bee-accessory.ts
@@ -98,6 +98,7 @@ export abstract class ZigBeeAccessory {
           this.coordinatorEndpoint
         );
         this.isConfigured = true;
+        this.log.info(`Device ${this.name} successfully configured!`);
       } catch (e) {
         this.log.error(`Cannot configure device ${this.name}: ${e.message}`);
         this.isConfigured = false;
@@ -105,9 +106,10 @@ export abstract class ZigBeeAccessory {
     }
   }
 
-  update(device: Device, state: DeviceState) {
+  async update(device: Device, state: DeviceState) {
     Object.assign(this.accessory.context, { ...device });
     Object.assign(this.state, state);
     this.zigBeeDeviceDescriptor.updateLastSeen();
+    await this.configureDevice();
   }
 }

--- a/src/index.ts
+++ b/src/index.ts
@@ -92,7 +92,11 @@ function registerSupportedDevices() {
   );
   registerAccessoryClass('Xiaomi', ['GZCGQ01LM'], XiaomiLightIntensitySensor);
   registerAccessoryClass('LUMI', ['lumi.weather', 'lumi.sensor_ht.agl02'], TempHumiSensor);
-  registerAccessoryClass('LUMI', ['lumi.sensor_magnet'], XiaomiContactSensor);
+  registerAccessoryClass(
+    'LUMI',
+    ['lumi.sensor_magnet', 'lumi.sensor_magnet.aq2'],
+    XiaomiContactSensor
+  );
   registerAccessoryClass('LUMI', ['lumi.sen_ill.mgl01'], XiaomiLightIntensitySensor);
   registerAccessoryClass('TuYa', ['GDKES-02TZXD'], TuyaOnoffDoubleSwitch);
   registerAccessoryClass('lk', ['ZB-MotionSensor-D0003'], LinkindMotionSensor);

--- a/src/platform.ts
+++ b/src/platform.ts
@@ -330,6 +330,7 @@ export class ZigbeeNTHomebridgePlatform implements DynamicPlatformPlugin {
       await this.initDevice(message.device);
     } else {
       this.log.warn(`Not initializing device ${ieeeAddr}: already mapped in Homebridge`);
+      await this.homekitAccessories.get(this.getAccessoryByIeeeAddr(ieeeAddr).UUID).onDeviceMount();
     }
   }
 

--- a/src/platform.ts
+++ b/src/platform.ts
@@ -77,8 +77,10 @@ export class ZigbeeNTHomebridgePlatform implements DynamicPlatformPlugin {
       channels.push(secondaryChannel);
     }
 
+    const port = this.config.port || (await findSerialPort());
+    this.log.info(`Configured port for ZigBee dongle is ${port}`);
     const initConfig = {
-      port: this.config.port || (await findSerialPort()),
+      port,
       db: this.config.database || path.join(this.api.user.storagePath(), './zigBee.db'),
       panId: this.config.panId || 0xffff,
       channels,

--- a/src/platform.ts
+++ b/src/platform.ts
@@ -11,7 +11,6 @@ import {
 
 import { PLATFORM_NAME, PLUGIN_NAME } from './settings';
 import { ZigBee } from './zigbee/zigbee';
-// import { RouterPolling } from './utils/router-polling';
 import * as path from 'path';
 import { findSerialPort } from './utils/find-serial-port';
 import { PermitJoinAccessory } from './accessories/permit-join-accessory';
@@ -34,6 +33,16 @@ import { Device } from 'zigbee-herdsman/dist/controller/model';
 const PERMIT_JOIN_ACCESSORY_NAME = 'zigbee:permit-join';
 const TOUCH_LINK_ACCESSORY_NAME = 'zigbee:touchlink';
 
+interface ZigBeeNTPlatformConfig extends PlatformConfig {
+  name: string;
+  port?: string;
+  panId?: number;
+  channel?: number;
+  secondaryChannel?: string;
+  database?: string;
+  routerPollingInterval?: number;
+}
+
 export class ZigbeeNTHomebridgePlatform implements DynamicPlatformPlugin {
   public readonly Service: typeof Service = this.api.hap.Service;
   public readonly Characteristic: typeof Characteristic = this.api.hap.Characteristic;
@@ -48,7 +57,7 @@ export class ZigbeeNTHomebridgePlatform implements DynamicPlatformPlugin {
 
   constructor(
     public readonly log: Logger,
-    public readonly config: PlatformConfig,
+    public readonly config: ZigBeeNTPlatformConfig,
     public readonly api: API
   ) {
     this.zigBee = new ZigBee(log);
@@ -203,18 +212,6 @@ export class ZigbeeNTHomebridgePlatform implements DynamicPlatformPlugin {
     this.client = new ZigBeeClient(this.zigBee, this.log);
     // Init devices
     this.zigBee.list().forEach(data => this.initDevice(data));
-    // Init log for router polling service
-    /*
-    if (!this.config.disablePingLog) {
-      const routerPolling = new RouterPolling(
-        this.zigBee,
-        this.log,
-        this.config.routerPollingInterval * 1000 || null
-      );
-      // Some routers need polling to prevent them from sleeping.
-      routerPolling.start();
-    }
-*/
   }
 
   private getAccessoryByIeeeAddr(ieeeAddr: string) {

--- a/src/platform.ts
+++ b/src/platform.ts
@@ -11,7 +11,7 @@ import {
 
 import { PLATFORM_NAME, PLUGIN_NAME } from './settings';
 import { ZigBee } from './zigbee/zigbee';
-import { RouterPolling } from './utils/router-polling';
+// import { RouterPolling } from './utils/router-polling';
 import * as path from 'path';
 import { findSerialPort } from './utils/find-serial-port';
 import { PermitJoinAccessory } from './accessories/permit-join-accessory';
@@ -165,7 +165,7 @@ export class ZigbeeNTHomebridgePlatform implements DynamicPlatformPlugin {
     if (!accessory) {
       // Wait a little bit for a database sync
       await sleep(1500);
-      this.initDevice(device);
+      await this.initDevice(device);
       return true;
     } else {
       this.log.debug(`Not initializing device ${device.ieeeAddr}: already mapped in Homebridge`);
@@ -204,6 +204,7 @@ export class ZigbeeNTHomebridgePlatform implements DynamicPlatformPlugin {
     // Init devices
     this.zigBee.list().forEach(data => this.initDevice(data));
     // Init log for router polling service
+    /*
     if (!this.config.disablePingLog) {
       const routerPolling = new RouterPolling(
         this.zigBee,
@@ -213,6 +214,7 @@ export class ZigbeeNTHomebridgePlatform implements DynamicPlatformPlugin {
       // Some routers need polling to prevent them from sleeping.
       routerPolling.start();
     }
+*/
   }
 
   private getAccessoryByIeeeAddr(ieeeAddr: string) {

--- a/src/utils/find-serial-port.ts
+++ b/src/utils/find-serial-port.ts
@@ -5,7 +5,6 @@ export async function findSerialPort(): Promise<string> {
     const ports = await serialport.list();
     const port = ports.find(item => item.manufacturer === 'Texas Instruments');
     if (port) {
-      console.log('Found COM port', port);
       return port.path;
     }
   } catch (e) {

--- a/src/utils/promise-queue.ts
+++ b/src/utils/promise-queue.ts
@@ -76,11 +76,8 @@ export abstract class PromiseBasedQueue<M, R> implements Queue<M, R> {
   processQueue(response: R): boolean {
     if (this.queuedMessages.length) {
       if (this.processResponse(this.queuedMessages, response)) {
-        console.log(`Message processed. Message queue size is now ${this.queuedMessages.length}`);
         return true;
       }
-    } else {
-      console.log('No message in queue, skipping');
     }
     return false;
   }
@@ -89,7 +86,6 @@ export abstract class PromiseBasedQueue<M, R> implements Queue<M, R> {
     const timestamp = new Date().getTime();
     const promise = new DeferredPromise<R>();
     this.queuedMessages.push({ timestamp, message, promise });
-    console.log(`Message queue size is ${this.queuedMessages.length}`);
     return promise.promise;
   }
 }

--- a/src/utils/router-polling.ts
+++ b/src/utils/router-polling.ts
@@ -7,7 +7,7 @@ export const DEFAULT_POLL_INTERVAL = 60 * 1000;
 export const MIN_POLL_INTERVAL = 10 * 1000;
 export const MAX_POLL_INTERVAL = 120 * 1000;
 
-function isDeviceRouter(device) {
+export function isDeviceRouter(device) {
   let power = 'unknown';
   if (device.powerSource) {
     power = device.powerSource.toLowerCase().split(' ')[0];

--- a/src/utils/router-polling.ts
+++ b/src/utils/router-polling.ts
@@ -3,7 +3,7 @@ import { ZigBee } from '../zigbee/zigbee';
 import Timeout = NodeJS.Timeout;
 import { Logger } from 'homebridge';
 
-export const DEFAULT_POLL_INTERVAL = 60 * 1000;
+export const DEFAULT_POLL_INTERVAL = 30 * 1000;
 export const MIN_POLL_INTERVAL = 10 * 1000;
 export const MAX_POLL_INTERVAL = 120 * 1000;
 

--- a/src/zigbee/zig-bee-client.ts
+++ b/src/zigbee/zig-bee-client.ts
@@ -28,7 +28,6 @@ export class ZigBeeClient extends PromiseBasedQueue<string, MessagePayload> {
     const deferredMessage = this.consumeMessage(key);
     if (deferredMessage) {
       this.log.debug(`Found message in queue for key ${key}, resolving`, response);
-      this.log.debug(`resolving promise`, deferredMessage.promise);
       deferredMessage.promise.resolve(response);
       return true;
     }

